packages/btcindexer/src/btcindexer.test.ts

```diff
@@ -478,3 +478,62 @@ describe("getSenderInsertStmts logic", () => {
 		expect(senderResults[0]!.sender).toEqual(fakeSenderAddress);
 	});
 });
+
+describe("Indexer.processFinalizedTransactions Front-run Detection", () => {
+	it("should detect when a tx was already minted by someone else and mark it as MINTED", async () => {
+		const blockData = REGTEST_DATA[329]!;
+		const txData = blockData.txs[1]!;
+
+		const db = await mf.getD1Database("DB");
+		await insertFinalizedTx(db, txData, blockData);
+
+		const kv = await mf.getKVNamespace("btc_blocks");
+		await kv.put(blockData.hash, Buffer.from(blockData.rawBlockHex, "hex").buffer);
+
+		vi.spyOn(indexer.nbtcClient, "tryMintNbtcBatch").mockResolvedValue(null);
+		vi.spyOn(indexer.nbtcClient, "isBtcTxMinted").mockResolvedValue(true);
+
+		await indexer.processFinalizedTransactions();
+
+		const { results } = await db
+			.prepare("SELECT * FROM nbtc_minting WHERE tx_id = ?")
+			.bind(txData.id)
+			.all();
+		expect(results.length).toEqual(1);
+		expect(results[0]!.status).toEqual("minted");
+		expect(results[0]!.sui_tx_id).toBeNull();
+	});
+
+	it("should handle partial front-running in a batch", async () => {
+		const blockData = REGTEST_DATA[327]!;
+		const tx1 = blockData.txs[1]!;
+		const tx2 = blockData.txs[2]!;
+
+		const db = await mf.getD1Database("DB");
+		await insertFinalizedTx(db, tx1, blockData);
+		await insertFinalizedTx(db, tx2, blockData);
```

Comment:
add todo - we should be able to insert many txs in batch. (and create a new task and assign it to yourself).